### PR TITLE
[Internal][Executor] Add aggregation nodes error summary to batch result

### PR DESCRIPTION
**Pull request by PeiwenGaoMS**
# Description

Add aggregation nodes error summary to batch result

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/executor/unittests/batch/test_result.py [hunk 1]
```diff
@@ -162,3 +168,19 @@ def test_invalid_api_calls(self, api_call):
         assert batch_result.system_metrics.total_tokens == 0
         assert batch_result.system_metrics.completion_tokens == 0
         assert batch_result.system_metrics.prompt_tokens == 0
+
+    def test_get_aggr_error_summary(self):
```
Comment (thy09) [inline src/promptflow/tests/executor/unittests/batch/test_result.py:172]: How about e2e tests?

File: src/promptflow/promptflow/batch/_result.py [hunk 2]
```diff
@@ -138,6 +138,7 @@ class BatchResult:
     metrics: Mapping[str, str]
     system_metrics: SystemMetrics
     error_summary: ErrorSummary
+    aggr_error_summary: Mapping[str, Any]
```
Comment (huaiyan) [inline src/promptflow/promptflow/batch/_result.py:141]: How about adding `aggr_error_list` in `ErrorSummary`?
Not sure if it's a good idea to have 2 error related summaries here.

File: src/promptflow/tests/executor/unittests/batch/test_result.py (hunk 1 above)
Comment (PeiwenGaoMS) [reply src/promptflow/tests/executor/unittests/batch/test_result.py:172]: After @codingrabbitt1 completes this task (https://msdata.visualstudio.com/Vienna/_workitems/edit/2699345/?view=edit), we can add an e2e test. Now if an aggregation node fails, an exception will be thrown.

File: src/promptflow/promptflow/batch/_result.py (hunk 2 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/batch/_result.py:141]: The current `error_summary` points more to the line error summary, I have 2 proposals:
1. `line_error_summary` and `aggr_error_summary` are in the first level of the batch result.
2. Move `aggr_error_summary` to `error_summary` and refactor `error_summary` to 
image

@huaiyan @thy09 which one do you prefer?

File: src/promptflow/tests/executor/unittests/batch/test_result.py (hunk 1 above)
Comment (PeiwenGaoMS) [reply src/promptflow/tests/executor/unittests/batch/test_result.py:172]: Added.

File: src/promptflow/promptflow/batch/_result.py (hunk 2 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/batch/_result.py]: Offline synced, we will use proposal 2.

**Review by huaiyan — DISMISSED** [review]

**Review by thy09 — APPROVED** [review]

**Review by Hhhilulu — APPROVED** [review]

**Pull request merged**